### ISSUE TITLE
Explore shared data between WordPress accounts

### ISSUE THREAD

**Issue by nico-martin** [google/site-kit-wp#6]
At the moment I'm only able to see the data connected to my own google account. I can't share data between other WordPress accounts.

It would be great to be able to share data with other WordPress accounts. I imagine having a list of all WordPress Users where I could select some Users that are able to see what I am seeing. Maybe they could then choose my Google account on the dropdown on the upper right corner.

This way it would also be possible to reveal the data to Admins that don't have a google account or if you have employees with a WordPress login (company-email) which is not a google account and you don't want to give access to the google services over their private google accounts. 

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Changelog entry

* 

## Acceptance criteria

*

**Comment by ThierryA** [comment]
Thanks for you suggestion Nico, nice to have you on this repo too 😄 

We having been discussing data sharing and I definitely see value and a strong use case for this feature. We have to get it right though, as data sharing may be sensitive.

I labelled this issue as **enhancement** for future versions, there is not ETA or guaranty though.

**Comment by nico-martin** [comment]
Hi Thierry,
Yes, I totally agree! That's a super sensitive topic :).
I'm looking forward to see how this plugin evolves!

**Comment by elyobo** [comment]
Agreed that this would be very useful.

* Other users may not have a Google account (and may not want one, nor need one for any other purpose).
* Other users may not be technically capable enough to feel comfortable setting up their own access.
* Requiring each user to set up their own access does not scale well; a site may have many users, and each having to set up their own access is much more painful than only one user setting it up.

While granular access (manually assign individual users access) would be handy, it's not the minimum implementation.  In terms of access control the following would use useful

* globally accessible (anyone may view these stats)
* role based access (anyone with role X may view these stats)
* user based access (user X may view these stats)

**Comment by mpitre72** [comment]
Food for thought... I've used or tested at least 10 other WP Google analytics plugins and they all allow the sharing of analytic data between user of WP.  I can't use Sitekit until you guys get this working to industry standards.  If you want to make it a setting where the default is "do not share" (even by connection like Analytics, search console, etc.) and then they can either choose specific WP users or access levels I totally get that, but please place this on your highest priority list.  It's an ineffective tool without sharing IMHO.

**Comment by billzyfem** [comment]
It should be up to the Wordpress admin who connected their data to decide which user roles gets to see it using the admin panel.

**Comment by gounder** [comment]
I strongly disagree with this suggestion, this is the only reliable Google Analytics plugin I have found which implements this feature this way. 

Others just share the data with all admin regardless of them having access to those reports on the actual Google Search Console and Google analytics dashboards, this way only those admin who already have access on Google will be able to connect and see reports.

The idea of seeing data in the dashboard is only a matter of convenience but the actual access controls managed from the Search Console and Analytics should be valid. 

---
In a clients organization we deactivated the backend dashboard feature as the client didn't want everyone to have access, but a manager who didn't have analytics access, used to activate and view reports and then deactivate the dashboard.

**Comment by elyobo** [comment]
@gounder it seems like having access controls within this plugin would allow you to have it whichever way you want it.

**Comment by mpitre72** [comment]
@gounder I have to second the comment made by @elyobo access controls by role or specific users makes perfect sense.  I don't see any reason why you would disagree with roles/user permissions.  If you are that hung up on access restrictions, then why even install dashboards in WP in the first place. 
 It seems your users should just have to work in the various Google products directly.

**Comment by billzyfem** [comment]
I don't think gounder understands this problem well enough and why it's important to have the admin of WP control who sees the data.  Until it's fixed that issue is a deal breaker for me otherwise we get clients calling saying 'google is set up wrong on my wordpress' (even though it isn't) and honestly who has time for these unnecessary calls to explain to every client that Google hasn't built the plugin right for them to see the data without extra configuration.  It makes the end result look incomplete and unfinished to the end user.

The average end user/client and their staff want their end result finished and complete. Most of them are not willing to click configuration buttons and connect service buttons.

**Comment by elyobo** [comment]
A good implementation of access controls would also support @gounder's use case; not sharing the data at all between accounts, as is currently done, should definitely remain an option as well.

**Comment by marrrmarrr** [comment]
Thanks everyone for the comments, great to see the interest and the lively discussion here. Given that one of the main goals of Site Kit is to make it as easy as possible to view the most relevant stats from Google in one place, I agree this would be a very useful feature. That said, we want to be very careful with how we store and share your Google data. 

If you share some more details about the use-cases you have in mind, it will help us understand what we can design to address the need while observing the security and identity requirements for your data. 
For example: 
- would you want to share per module (e.g. AdSense only or SC only), or the entire dashboard at once?
- would you want to share the entire site, or individual posts (e.g. if you have contributors authoring content in specific sections, they only see stats for their content)? 
- would you want to share per role type (e.g. all editors) or only with specific individuals? 
- would the ability to generate and share e.g. a monthly report of site performance with any user on the WP site (or via email) address this need?
- any other details about the example use-cases do you have in mind?

**Comment by razorfrog** [comment]
This is the main feature holding us back from installing on 150+ client sites. Responding to @marrrmarrr above: 

- would you want to share per module (e.g. AdSense only or SC only), or the entire dashboard at once? **Analytics is most important for us**

- would you want to share the entire site, or individual posts (e.g. if you have contributors authoring content in specific sections, they only see stats for their content)? **Entire site**

- would you want to share per role type (e.g. all editors) or only with specific individuals? **Entire role would be fine - could then use a user-role editor plugin to fine tune if necessary**

- would the ability to generate and share e.g. a monthly report of site performance with any user on the WP site (or via email) address this need? **No, would like to see it on the dashboard**

**Comment by billzyfem** [comment]
From our end, we're similar like Razorfrog scenario, lots of sites we cannot move over to site kit until it works in a wordpress friendly way.

would you want to share per module (e.g. AdSense only or SC only), or the entire dashboard at once? **Googe Anaytics dashboards inside WP admin is the most important part.**

would you want to share the entire site, or individual posts (e.g. if you have contributors authoring content in specific sections, they only see stats for their content)? **Entire site generally, although having the option of content specific would be useful in some scenarios**

would you want to share per role type (e.g. all editors) or only with specific individuals? **Roles & Capabilities would be perfect.  Most savvy admins have their own permissions editor plugins installed for more advanced implementations.**

would the ability to generate and share e.g. a monthly report of site performance with any user on the WP site (or via email) address this need? **Not really.**

any other details about the example use-cases do you have in mind? **These are more long term ideas that would make the plugin superior once the above is addressed and the kind of thing I'd expect Google to do better than any other developers;  Proper WooCommerce ecommerce tracking integration, easier ways to pass data from certain posts/pages back into analytics ie standard events, improved google tag manager integration to pass data through the container for certain conditions and an easier solution to address cross domain tracking**

**Comment by svensonsan** [comment]
Hi 

I recently stumbled across this issue with 60+ people on our company site not being able to see analytics on the dashboard. 

- would you want to share per module (e.g. AdSense only or SC only), or the entire dashboard at once? **Per Module. Adsense only for admins, Analytic & Search Console for all editors** 

- would you want to share the entire site, or individual posts (e.g. if you have contributors authoring content in specific sections, they only see stats for their content)? **Share the complete site** 

- would you want to share per role type (e.g. all editors) or only with specific individuals? **Yes. Per User Role**

- would the ability to generate and share e.g. a monthly report of site performance with any user on the WP site (or via email) address this need? **Sharing means here I have to take action? Then no, users should be able to do this for themself.**

- any other details about the example use-cases do you have in mind? Mention this limitation prominent on the wordpress plugin page. **Currently this plugin doesn't work with multiple editors across the globe in a company. It's great for one person, but doesn't work for >1 amount of editors. Even if people have access to search console and analytics the setup if way, really totally too much for a regular content writer. **

**Comment by anywarewebmarketing** [comment]
We need this feature too. I have clients asking for access to their Google Analytics data from the plugin they can see in WordPress, but when they click it they are presented with an install process they do not understand and should not have to go through (because we are supposed to do all this technical stuff for them).


    would you want to share per module (e.g. AdSense only or SC only), or the entire dashboard at once? entire dashboard for me, but modular would be fine too
    would you want to share the entire site, or individual posts (e.g. if you have contributors authoring content in specific sections, they only see stats for their content)? entire site
    would you want to share per role type (e.g. all editors) or only with specific individuals? individuals but roles would be ok too and is probably more practical to manage
    would the ability to generate and share e.g. a monthly report of site performance with any user on the WP site (or via email) address this need? possibly but it would need to be as good as the scheduled reports built into Google Analytics. We're happy to continue using the GA scheduled reports.

Thanks!

**Comment by tlhInganHom** [comment]
Just chiming in that I also need the ability for my site's users to see the analytics data from within wordpress. I run a news site with around 50 different authors publishing posts, and they like to track how their posts are doing. Even just a separate dashboard widget that would show them this information would be really grand.

**Comment by timiwahalahti** [comment]
👋 Site Kit is the plugin we have been waiting for to pop up, all other plugins with similar functionalities are just pain in the ass. But like others in the thread, the lack of sharing of data is holding us back from installing the plugin for our 100+ clients.


- would you want to share per module (e.g. AdSense only or SC only), or the entire dashboard at once? **Analytics is by far most important information, but SC, AdSense and Tag Manager sharing would be sometimes useful too.**
- would you want to share the entire site, or individual posts (e.g. if you have contributors authoring content in specific sections, they only see stats for their content)? **Without a question; the whole site, as it would allow setting up only once and share data with the whole marketing department.**
- would you want to share per role type (e.g. all editors) or only with specific individuals? **If I had to choose, per role type.**
- would the ability to generate and share e.g. a monthly report of site performance with any user on the WP site (or via email) address this need? **No.**

**Comment by Alex-Hay** [comment]
I'd like to quickly echo how useful this feature would be. We manage sites for a few dozen sites, all of which link up to Google Analytics. We've become increasingly frustrated at the price and questionable ethics of companies like Monster Insights, but unfortunately the lack of even a basic dashboard widget (for editors) makes Site Kit a non starter.

Not trying to seem ungrateful though - the plugin is great otherwise!

**Comment by Buildbetter** [comment]
Please correct this is very bad mistake. Make it option, give access to all administrators to see statistics. Its very bad decision. Please fix it, don't kill us.

**Comment by grpace** [comment]
As an agency that sets up analytics for our clients. It is imperative that our client user role with different accounts don't need to log into analytics but just get the authenticated access that we have set up on our admin account.

Please give us this option!

**Comment by GPurdue** [comment]
Hi there, just another agency that would need to set this up on 50+ clients sites and then have to get each client to create a Google Analytics account, grant them access to their analytics and then get them to sign into the Site Kit on their site with their analytics account.

I would prefer a role checkbox that would allow me to allow any other admin users to see the analytics without the need for them to sign into their own analytics account.

**Comment by TaisaDesigner** [comment]
I think this should be maybe not default but as an option.

My use case:

I am a web-designer. My customers are quite basic with technology sometimes, and ask them to connect their google account is a big issue for them. Sometimes I even create the Analytics and Search console for them and pass on them later. So I think it would be great to let this connected to their sites somehow and that they don’t need to do anything else.

Maybe an option like "share the details with other admins"? Or choose some users to share the info with?

**Comment by nicosemp** [comment]
> * would you want to share per module (e.g. AdSense only or SC only), or the entire dashboard at once?
Sharing the entire dashboard would be the basic functionality. I haven't had any clients who required a per module sharing.
> * would you want to share the entire site, or individual posts (e.g. if you have contributors authoring content in specific sections, they only see stats for their content)?
Entire site. I don't see ever needing to exclude a custom post or a set of pages from a user's view.
> * would you want to share per role type (e.g. all editors) or only with specific individuals?
Per role would be great.
> * would the ability to generate and share e.g. a monthly report of site performance with any user on the WP site (or via email) address this need?
This is not fundamental, but would be a nice to have addition.

**Comment by finalhit** [comment]
Shouldn't this be a top priority feature request? All webmasters and agencies need this for their clients.

**Comment by ThierryA** [comment]
Thanks all for you comments and sharing suggestions.

This issue is definitely in our priority list and we are discussing the security and privacy implications. To be clear, the complexity here is not from an engineering perspective and **not the time it would take to implement data sharing**, it is definitely due to security and privacy.

As for today, allowing a WP user to **request** data on **behalf** of another Google account is definitely a **no go** from a security and privacy perspective. As some of you mentioned, other non Google popular plugins may do that, however this is **not a good reason for Site Kit** to do the same 😉 

With all that said, I would like to pivot this discussion to see this feature from a differ angle. We could/should look at it as **sharing a view/snapshot** and not requesting data on behalf of someone else. The end result is pretty similar but the underlying approach to the discussion and engineering implementation is fundamentally different because no request would be made on behalf of someone. Instead the idea is that a snapshot of the dashboard is regularly taken and accessible by the people it is is shared with. It is worth noting that it is still early days for this approach and we still have a **long way to go** to confirm whether this is a **feasible approach or not**.

**Comment by reaxion** [comment]
I'm well on my way to having this plugin installed in 150+ WP sites before discovering this problem.

A previous free plugin gone commercial showed analytics to all users without problems for years - maybe send them a DM?

**Comment by nico-martin** [comment]
If I understand it correctly, this plugin allows a lot more than previous analytics plugins did. Thats why you need to authenticate as a user insteand of an access key with limited access rights.
With this in mind it's understandable that they really need to take the time and find a solution that would not just open up any dangerous side-effects.

Logging into other user's accounts and signing in with your Google account is a pretty bad idea. And that's definitely not what the authors would suggest. If you really need to share the data between accounts you should consider using a different plugin.

**Comment by bethanylang** [comment]
@seriouslyjennings Thanks for sharing your use case and interest in this feature. This is definitely still on our radar as an improvement to Site Kit, but as this time, we don't have a detailed timeline. Please feel free to follow this issue to receive future updates. Thank you!

**Comment by ThierryA** [comment]
Quick update on this thread, this issue has now been prioritized and we are actively exploring solutions. I don't want to set false expectation though, this could take quite some time to 1. find a solution 2. validate the solution with our security team 3. build it. So no ETA and no guaranty at this time but I hope sharing that we are actively exploring solutions is already kinda good news 😄

**Comment by marrrmarrr** [comment]
Hi everyone, thanks for sharing your suggestions and describing your top use cases in detail. 
As we're discussing the potential solutions for this, there's a question we'd appreciate your feedback on -- if all that was necessary for you to share the dashboard with other users of the site is that they log in to their Google account (no need for them to be a WP admin or to have access to Search Console, GA, etc) would that solve your use case?

**Comment by nicosemp** [comment]
It most definitely would!
Are you thinking about a whitelist of Google accounts that, once logged in, have access to the shared dashboard?

**Comment by ivptr** [comment]
It would not solve the issue, because it requires Google Account for every user.

**Comment by elyobo** [comment]
Agreed, the problem is the Google sign in. It doesn’t scale and some
clients may not have or want an account.

In the same sense that I can produce a static report from GA and share it,
I should be able to share access to the data presented by site kit without
requiring the recipients to authenticate with Google.

**Comment by seriouslyjennings** [comment]
It would not solve mine automatically, but it would definitely be a step in the right direction. 
Here are a few scenarios to think about :

- I have some clients that are very not tech savvy, and they (arguably) shouldn’t be logging into the dashboard anyway, but they want to be able to see the stats. When it asks for their google account & password, they think it’s broken and think I didn’t set it up. 
- The email they use to log in to the site might not be a gmail account, so will we be able to whitelist people from outside of gmail?  

Ideally it would be something where I can control who sees the stats by user role, and my initial login would be enough to hook it all up so they could just click google site kit and see the analytics. 

Thanks!
-Sarah

**Comment by timiwahalahti** [comment]
> if all that was necessary for you to share the dashboard with other users of the site is that they log in to their Google account (no need for them to be a WP admin or to have access to Search Console, GA, etc) would that solve your use case?

It would not. As few others have already pointed out, not everyone has a Google account.

This would also add an extra step to see the analytics, which is bad. I want to be able to say to our customers that they can see the quick glimpse straight from the WP dashboard.

**Comment by GPurdue** [comment]
> Hi everyone, thanks for sharing your suggestions and describing your top use cases in detail.
> As we're discussing the potential solutions for this, there's a question we'd appreciate your feedback on -- if all that was necessary for you to share the dashboard with other users of the site is that they log in to their Google account (no need for them to be a WP admin or to have access to Search Console, GA, etc) would that solve your use case?

I'm sorry but as others have said, this wouldn't help. Many of my clients don't have Google accounts or don't have a Google account associated with their business. The ideal scenario would be that I create their GA account, link it to the site using the Site Kit plugin and I can choose which users or roles can view the analytics, then when they login, they see their analytics.

**Comment by nlenkowski** [comment]
Hi. Jumping in a bit late here, but I've run into this same issue on nearly every client site I've installed Site Kit on. Clients don't read the "Connect your Google account to Site Kit" message and think the plugin has been misconfigured. When I explain the situation they're often frustrated that Site Kit doesn't work like their "old" analytics plugin, and that gaining access to the dashboard analytics is so complicated.

I agree with @GPurdue completely. Once a Google Account has been linked to Site Kit then access to the Search Console and Analytics dashboards should be configurable by WordPress user/role, and not tied to a Google account in any way.

Thanks @ThierryA for considering this use case, and for your continued work on Site Kit of course! 😄

**Comment by billzyfem** [comment]
It's been 2 long years watching this thread, and we're still no closer to being able to use Google Site Kit... Not getting my hopes up either.  It's simply a bad experience for clients the way it is currently.

**Comment by TaisaDesigner** [comment]
I don't really get what is the problem on sharing to other admins or specific users the ability to view the data of that web.

Anyway, I stopped using the plugin because it gave me some problems, and also because of the new cookie policies in Europe.

**Comment by Peter-HVD** [comment]
`<rant>`It's very disappointing to see that this issue has been dragging on for over 2 years ago. The problem is caused by a typical one that exists amongst plugin creators in that "I say no, because I know best" when, in reality, the best thing to do is to give the user the choice - yes, there may be issues giving access to this sort of data to certain user, but - to be blunt - that's our problem and not yours!`</rant>`

The best solution in my opinion is to add a number of custom capabilities - one for each module, for example `sitekit_view_adsense`, `sitekit_view_pagespeed`, `sitekit_view_analytics`, etc, and reserving the admin functionality to the usual `manage_options` - so that the site's admin can then grant those capabilities to whomever s/he chooses.

**Comment by marrrmarrr** [comment]
Hey @Peter-HVD and everyone who provided feedback so far, thanks for the detailed info on the various use cases and your needs. It's been very helpful for us to scope the minimum viable features. 
It did take a while to define exactly how to build this capability while making sure we're in line with privacy and security requirements. That said, right now this feature is a top priority and we're working actively on it. We do have an Epic for it already (#3048), expect more GH issues popping in there in the near future (:

**Comment by marrrmarrr** [comment]
Hi @Alex-Hay, @anywarewebmarketing, @billzyfem, @GPurdue, @grpace, @nico-martin, @nicosemp, @Peter-HVD, @razorfrog, @seriouslyjennings, @svensonsan, @TaisaDesigner, @timiwahalahti (and anyone else I might have missed who's shared feedback so far on this topic) -- thanks again for sharing your thoughts, we've made a ton of progress on designing the core dashboard sharing flows and will be starting development soon 🙌

We're now going through some corner cases, so we'd love to understand more specifically your thoughts on the "second admin" use-case: 
- Would you be sharing the view-only dashboard with a second (or third etc) admin after doing the initial setup? Or would you be sharing primarily with non-admin roles (editors, etc)?
- Do most of your clients / end users of sites you manage have "admin" as WP role?

Thanks again for being vocal and helpful about this feature!

**Comment by douglaskarr** [comment]
> * Would you be sharing the view-only dashboard with a second (or third etc) admin after doing the initial setup? Or would you be sharing primarily with non-admin roles (editors, etc)?

As the agency or consultant, we typically have administrative rights so that we can manage the site in its entirety. Our clients typically have editorial rights to write and publish but not change actual site settings. So... we'd set up Site Kit and connect the accounts. Ideally, we'd like them to see the reports but not be able to disable / enable disconnect / connect other google accounts.

> * Do most of your clients / end users of sites you manage have "admin" as WP role?

No, we typically do have one admin since they own the site; however, the rest of the staff is typically an editor, contributor, or author. We do this so that they avoid messing up any settings or configurations in WordPress. Ideally, I think listing out the roles and being able to enable or disable (1) connecting accounts and (2) reading reports would be ideal. I think some people may even want contributors to be able to read the reports could be a possibility.

**Comment by elyobo** [comment]
Agreed with @douglaskarr  above; we have a network of ~900 sites that we don't give admin to, but would like to share data to editors.

**Comment by billzyfem** [comment]
For us the permissions vary per project.  What would be best is if we can select the roles that are able to view the reports.  One admin connects to google and then the rest can see the reports if their role allows it without needing to connect to google directly.

**Comment by nicosemp** [comment]
We are very happy this feature is being worked on.

> * Would you be sharing the view-only dashboard with a second (or third etc) admin after doing the initial setup? Or would you be sharing primarily with non-admin roles (editors, etc)?

With other plugins, one of the admins makes the connection, and shares the view with other (but not necessarily all) developers and the client. So yes we do need to be able to share it with some of the other admins.

> * Do most of your clients / end users of sites you manage have "admin" as WP role?

They usually have the "editor" role, but not always. We have some clients with the "admin" role for various reasons.

Overall, having the options to enable the view both per user and per role would be perfect. If that is not possible, having one or the other option would also work for us.

**Comment by GPurdue** [comment]
> * Would you be sharing the view-only dashboard with a second (or third etc) admin after doing the initial setup? Or would you be sharing primarily with non-admin roles (editors, etc)?

I've setup a 'Client' role that most of my clients use. Some use the admin role if I know they can be trusted not to mess the site up. So from my point of view, a list of all roles that can be selected to view the dashboard would be perfect.

> * Do most of your clients / end users of sites you manage have "admin" as WP role?

No, some do, some don't.

Thank you for working on this issue, I'm looking forward to seeing what you come up with. As nicosemp mentioned above 'Overall, having the options to enable the view both per user and per role would be perfect. If that is not possible, having one or the other option would also work for us.'

**Comment by timiwahalahti** [comment]
Sorry for coming a little late with response here @marrrmarrr!


> Would you be sharing the view-only dashboard with a second (or third etc) admin after doing the initial setup? Or would you be sharing primarily with non-admin roles (editors, etc)?

This varies somewhat. To simplify, typically we set up the tracking for clients and they use it as view-only.

> Do most of your clients / end users of sites you manage have "admin" as WP role?

Most of them do have the admin role, yes, at least the "project owner" from the client has and others might have lower capabilities.

**Comment by nesoor** [comment]
What's the status of this topic ?
Are there plans to make sharing available because I just wanted to replace Monsters insights with Google Site Kit but I immediately noticed that other user roles couldn't see the data which is a bummer.

**Comment by bethanylang** [comment]
Hi @nesoor! We're actively working on a plan for sharing data, but don't have a timeline for a release at this time. Keep following here for updates.

**Comment by nesoor** [comment]
> Hi @nesoor! We're actively working on a plan for sharing data, but don't have a timeline for a release at this time. Keep following here for updates.

Thanks for the reply.
Looking forward to it ! :)

**Comment by ted81das** [comment]
Great to know this is work in progress. Please keep us posted on the updates.

**Comment by JeffBurris** [comment]
I just want to show my support for this. I was so hoping my lesser-techy partner who I wrote the website for and who edits products, orders etc. in Wordpress could just at least see the initial pie graph of growing visitors per day without making him set up all over again or me walk him through using my gmail... for now I'm just sending him a screenshot once a day of the pie graph. Thanks so much!

**Comment by CClanda** [comment]
I have the same problem! The plugin is practically USELESS if you can't show analytics to users with lower privileges. I tried with other methods to add the permission without success.

Please fix this!!!

**Comment by bethanylang** [comment]
Hi all! Thank you again for your patience and ongoing feedback on this issue – we know that this has been a long time coming.

Right now, our plan is to begin engineering on this issue in early 2022, with a release in Q2 or Q3 2022. We'll continue to update this issue as more details become available, so stay tuned!

**Comment by Picpool** [comment]
HI,
Actually i am pretty surprised that this main feature is not in Google Site Kit.
Maybe have i missed something....
Because like with plugins as https://fr.wordpress.org/plugins/google-analytics-dashboard-for-wp/ 
users need to see the analytic dashboard without having to log with multiple accounts.

**Comment by henryavila** [comment]
Any news about this?
I have a dissatisfied client because they don't have access to analytics

**Comment by bethanylang** [comment]
@henryavila Thanks for getting in touch! We're actively working on a new feature that will allow admins to share access to the Site Kit Dashboard with non-admins. This is still in the planning stages, but right now we're aiming to release it in Q4 2022.

**Comment by SurajVerma** [comment]
Any update on this ? This is creating problems for my installation as well.

**Comment by bethanylang** [comment]
@SurajVerma Thanks for getting in touch! We’re making great progress on the Dashboard Sharing feature, which will allow administrators to share access to the Site Kit dashboard with non-admin users. Unfortunately, we’ve encountered some blockers that are impacting our ability to set a release date for this functionality in Site Kit. We’ll share another update on this issue as soon as we have more information about a timeline.

**Comment by w0rldart** [comment]
Even admin to admin suffers from the same issue.

What's the point of using this plugin if my admin colleagues cannot see what I have configured?

**Comment by jamesozzie** [comment]
@w0rldart Multiple administrators can set up and use Site Kit currently. This has always been the case, with more information on this below:
https://sitekit.withgoogle.com/documentation/getting-started/install/#additional-users 

Note that if one administrator sets up Site Kit and connects services such as Google Analytics or AdSense, they will need to provide access to the connected Google Analytics property or AdSense account to any other administrators. Without doing so, any additional administrators who sets up Site Kit may encounter permission notices, if access was not shared. 

If you encounter any problems or have questions on this feel free to [open a support topic for assistance](https://wordpress.org/support/plugin/google-site-kit/#new-post).

**Comment by VR51** [comment]
@jamesozzie

Re "[Without being given permission to access existing Google Services accounts], any additional administrator who sets up Site Kit may encounter permission notices, if access was not shared."

That can lead to Google registering other administrators as the site owner, the creation of new Google services properties for the site eg new GA accounts or new Search Console accounts, and the original administrator's Google Services getting disconnected from the site which in turn means the real owner of the site can no longer view stats for their own site until Site Kit is reconfiguration under the real owner's Google Services account.

It can get a bit messy.

**Comment by douglaskarr** [comment]
Administrative access should always afford the opportunity to manage the plugin's connections. That's what full administrative rights are. The best practice for all plugins should be:

- Notification to administrators if any administrative settings have been changed in the plugin.
- A role and permissions settings page that allows the predefined roles in WordPress to View the reports. Perhaps I wish to give permission to Editors, Authors, and Contributors to view the reports. None would be able to change the connection to the services, but could view the site kit reporting.

This wouldn't be sharing access within Google's properties... the plugin should only provide access to the connectivity to those services.

**Comment by JonathanCrossland** [comment]
Is this done yet? I have same issues and trying to resolve.
Not everyone is a developer/ or knowledgeable of how to connect accounts etc - 
Either the users declared in google analytics should just have access - or wordpress users/permissions.

**Comment by billzyfem** [comment]
We've been waiting for 3 years (literally).  This is never going to be resolved :(

**Comment by nesoor** [comment]
> JonathanCrossland

Nope and as you can see this post is from 2019 so don't expect much from the Google Development team.
The plugin will probably end up in their graveyard of projects.

**Comment by martin-levai99** [comment]
This is something I would appreciate. I hope this will be resolved...

**Comment by Julian367** [comment]
Any ETA for this?

**Comment by ThierryA** [comment]
There is no precise ETA but I can share that we should very close to being able to launch this (assuming we don't get blocked by the few last items).

This feature has been without doubt the most demanding from a UX and engineering perspective but more importantly from a security and privacy perspective which has been a significant challenge to do **properly**. Transparently, we went back to the drawing board countless times to find an elegant solution which complies with security and privacy strict requirements. We got blocked multiple times along the way but at this point I _think_ the path forward is cleared 😉 

I can't promise any date as we still have a few things on our plate before launching this, we will definitely make sure to update this issue once we start rolling it out.

Stay tuned,

**Comment by bethanylang** [comment]
Hi all! I am thrilled to share that the Dashboard Sharing feature is now available! This will be an incremental rollout, so you may not see the feature immediately. If you can't see the Dashboard Sharing feature just yet, don't worry – it will be coming to you soon.

More information on Dashboard Sharing can be found on our [latest blog post](https://sitekit.withgoogle.com/news/share-the-site-kit-dashboard-with-your-team/) as well as our [support guide](https://sitekit.withgoogle.com/documentation/using-site-kit/dashboard-sharing/).  If you have any questions on this feature then please open a new topic on our [support forum](https://wordpress.org/support/plugin/google-site-kit/), where we will be happy to help.

**Comment by marrrmarrr** [comment]
Thanks everyone for the feedback and feature suggestions. 
If you have other use cases or any more feedback, we'd love to hear about it in the [forum](https://wordpress.org/support/plugin/google-site-kit/), as @bethanylang mentioned already.  

It took a while, but I'm happy to close this issue now. Enjoy sharing the dashboard with your team or customers (-:

**Comment by razorfrog** [comment]
Hooray! Can you define "soon" for the rollout? I don't see it yet but I'd love to test it out and consider adding the plugin back to all of our client sites. Thank you.

**Comment by marrrmarrr** [comment]
@razorfrog right now it's rolled out to 10% of sites, we will keep it like that likely for a couple of weeks or so to assess usage and any potential unforeseen corner cases. If everything looks good, we will open up to more sites.

**Comment by eddieali** [comment]
Thank you all for your work on the new Dashboard sharing feature. I can see it on many websites we manage and already utilizing the ready-only dashboard feature. Do we have a timeline for when this feature will be available for everyone, including new installations?

**Comment by jamesozzie** [comment]
@eddieali We don't have an exact timeline on the feature being fully rolled out, however, we do expect this to be happening soon. Right now it is available for 50% of Site Kit users, with this number set to increase over the next few weeks. 

> Sorry, I am not sure if it is too much to ask, but can we also enable "Site Kit Summary" for view-only editors and other administrators? Currently, the summary only shows on the main admin account.

Great question! The "Site Kit Summary" widget which appears on the main WordPress dashboard for administrators, is not part of Dashboard Sharing. At present it will only appear for site administrators. I will, however, pass your query on to the team regarding this.

**Comment by dubpir8** [comment]
> @eddieali We don't have an exact timeline on the feature being fully rolled out, however, we do expect this to be happening soon. Right now it is available for 50% of Site Kit users, with this number set to increase over the next few weeks.
> 
> > Sorry, I am not sure if it is too much to ask, but can we also enable "Site Kit Summary" for view-only editors and other administrators? Currently, the summary only shows on the main admin account.
> 
> Great question! The "Site Kit Summary" widget which appears on the main WordPress dashboard for administrators, is not part of Dashboard Sharing. At present it will only appear for site administrators. I will, however, pass your query on to the team regarding this.

It's March now. Any update on when this will be available for all users?

**Comment by dubpir8** [comment]
It's March now. Will this be updated or will it take another 3 years maybe? lol

**Comment by mmcwerbung** [comment]
How can i show the google site kit to other user groups?

**Comment by jamesozzie** [comment]
@mmcwerbung You can share a view only version of the Site Kit dashboard to non administrator roles using the Dashboard Sharing feature. We have a [guide on the plugin website](https://sitekit.withgoogle.com/documentation/using-site-kit/dashboard-sharing/) in relation to this.

If this feature isn't available to you at this time, note the [comment directly above](https://github.com/google/site-kit-wp/issues/6#issuecomment-1451598764), with a mention that all users will have access once it's rolled out to 100% of users, at some stage today or tomorrow. 

For any further questions on this, feel free to [open a support topic](https://wordpress.org/support/plugin/google-site-kit/#new-post) on the plugins WordPress support forum.

**Comment by jamesozzie** [comment]
@mmcwerbung Just to let you know that the feature has now been rolled out to all users. Please check whether you have this feature enabled by reloading your Site Kit dashboard and checking for this feature. You'll find [details on the plugin website](https://sitekit.withgoogle.com/documentation/using-site-kit/dashboard-sharing/#usingdashboardsharing) on how to use Dashboard Sharing. If you encounter any problems or queries please [open a support topic](https://wordpress.org/support/plugin/google-site-kit/#new-post) and we'd be happy to assist.
